src/test/java/com/adaptavist/tm4j/jenkins/cucumber/CucumberReportParserTest.java

```diff
@@ -0,0 +1,50 @@
+package com.adaptavist.tm4j.jenkins.cucumber;
+
+import org.apache.commons.io.FileUtils;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.PrintStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+public class CucumberReportParserTest {
+
+
+    public final static String TMP_FOLDER = "tmp/";
+
+    @Test
+    public void shouldFilterFile() throws Exception {
+
+        File oldFile = new File("src/test/resources/outputFiles/result_1.json");
+        File newFile = CucumberFileUtil.filterCucumberFiles(oldFile, TMP_FOLDER, true, System.out);
+
+        String data = FileUtils.readFileToString(newFile, "UTF-8");
+        String data2 = FileUtils.readFileToString(new File("src/test/resources/cucumber/result_1_filtered.json"), "UTF-8");
+        assertTrue(newFile.exists());
+        assertEquals(data, data2);
+        newFile.delete();
```

Comment:
Not a big deal since it's just for local tests but I suppose that if any assertion fail, `newFile` won't be deleted, right?

Reply:
fixed